src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,10 +26,27 @@ const ActorInit En_Fr_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fr/EnFr_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fr/EnFr_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fr/EnFr_Update.s")
+void EnFr_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFr* this = THIS;
+
+    if (Flags_GetSwitch(globalCtx, ENFR_GET_SWITCHFLAG(thisx))) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        this->actor.targetMode = ENFR_GET_TARGETMODE(thisx);
+    }
+}
+
+void EnFr_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnFr_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnFr* this = THIS;
+
+    if (Flags_GetSwitch(globalCtx, ENFR_GET_SWITCHFLAG(thisx))) {
+        Actor_MarkForDeath(&this->actor);
+    } else if (this->actor.xyzDistToPlayerSq < (gGameInfo->data[0x37D] * gGameInfo->data[0x37D])) {
```

Comment:
The `gGameInfo->data[0x37D]`s are `IREG(29)`. If you see `gGameInfo`, you can get the appropriate register by running tools/regconvert.py. And you can also use the `SQ` macro instead of multiplying twice.

```suggestion
    } else if (this->actor.xyzDistToPlayerSq < SQ(IREG(29))) {
```

Reply:
Thank you very much!